### ISSUE TITLE
Sphinx latex style file wrongly inhibits colours for section headings for latex+dvi(ps,pdf,pdfmx)

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2693]
Test document

``` rst
Hello
=====

Chapter
-------

section
^^^^^^^

section
^^^^^^^

section
^^^^^^^

section
^^^^^^^
```

and one can add

``` rst
.. raw:: latex

   \definecolor{TitleColor}{rgb}{1,0,0}
   \relax
```

for more dramatic impact. If using `make latexpdf` the section headings are coloured (and in red with extra raw code). But `make latex` and then using `xdvi` to check the `dvi` file shows no colour. Same naturally in final pdf if produced by `dvipdfmx` or by `dvips+ps2pdf`.

Culprit is strange

``` latex
\ifx\pdfoutput\undefined\else\ifcase\pdfoutput
  \let\py@NormalColor\relax
  \let\py@TitleColor\relax
\else
```

in sphinx.sty. But `dvi` files may perfectly well carry colour information, and  `xdvi` displays them. For example one can add a literal-block and customize `VerbatimBorderColor` to not be black. This is not inhibited by sphinx.sty, only `TitleColor` is.

This code-path appears faulty.

**Comment by jfbu** [comment]
more complete test:

``` rst
Hello
=====

.. raw:: latex

   \definecolor{TitleColor}{rgb}{1,0,0}
   \definecolor{VerbatimBorderColor}{rgb}{0,0,1}
   \relax

Chapter
-------

section
^^^^^^^

.. code-block:: python

   1+1

section
^^^^^^^

section
^^^^^^^

section
^^^^^^^
```

and a screenshot from xdvi:

![capture d ecran 2016-06-18 a 11 45 35](https://cloud.githubusercontent.com/assets/2589111/16170177/4ab8ae64-354a-11e6-8478-61dbd6ac6672.png)

the sections should be red, they are in pdf produced by `make latexpdf`.

**Comment by jfbu** [comment]
fixed as consequence of 4fb01ca
